docs/source/compatibility.rst

```diff
@@ -105,97 +91,95 @@ and how users enable or disable the warnings practically.
 
 Documentation Error
 ~~~~~~~~~~~~~~~~~~~
 
-Compatibility is basically determined based on the documentation, though it sometimes contains errors.
+Compatibility is basically determined based on the documentation, although it sometimes contains errors.
 It may make the APIs confusing to assume the documentation always stronger than the implementations.
 We therefore may fix the documentation errors in any updates that may break the compatibility in regard to the documentation.
 
 .. note::
-   Developers MUST NOT fix the documentation and implementation of the same functionality at the same time in revision updates as "bug fix".
-   Such a change completely breaks the backward compatibility.
-   If you want to fix the bugs in both sides, first fix the documentation to fit it into the implementation, and start the API changing procedure described above.
+
+   Developers should not fix the documentation and implementation of the same functionality at the same time in revision updates as a "bug fix"
+   unless the bug is so critical that no users are expected to be using the old version correctly.
 
 Object Attributes and Properties
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Object attributes and properties are sometimes replaced by each other at minor updates.
+Object attributes and properties are sometimes replaced by each other.
 It does not break the user codes, except the codes depend on how the attributes and properties are implemented.
 
 Functions and Methods
 ~~~~~~~~~~~~~~~~~~~~~
 
-Methods may be replaced by callable attributes keeping the compatibility of parameters and return values in minor updates.
+Methods may be replaced by callable attributes keeping the compatibility of parameters and return values.
 It does not break the user codes, except the codes depend on how the methods and callable attributes are implemented.
 
 Exceptions and Warnings
 ~~~~~~~~~~~~~~~~~~~~~~~
 
 The specifications of raising exceptions are considered as a part of standard backward compatibilities.
-No exception is raised in the future versions with correct usages that the documentation allows, unless the API changing process is completed.
+No exception is raised in the future revision versions with correct usages that the documentation allows.
 
-On the other hand, warnings may be added at any minor updates for any APIs.
-It means minor updates do not keep backward compatibility of warnings.
+On the other hand, warnings may be added at any revision updates for any APIs.
+It means revision updates do not keep backward compatibility of warnings.
 
 Model Format Compatibility
 --------------------------
 
-Objects serialized by official serializers that Chainer provides are correctly loaded with the higher (future) versions.
+Links and chains serialized by official serializers that Chainer provides are correctly loaded with the future versions.
```

Comment:
Do we ensure model compatibilities are kept even an **major** version up is done?